compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/optimizer/LargeMethodOptimizer.java

```diff
@@ -593,8 +610,8 @@ private BIRFunction createNewBIRFunctionAcrossBB(BIRFunction parentFunc, Name fu
         }
         BInvokableType type = new BInvokableType(paramTypes, retType, null);
 
-        BIRFunction birFunc = new BIRFunction(parentFunc.pos, funcName, funcName, parentFunc.flags, type,
-                parentFunc.workerName, 0, parentFunc.origin);
+        BIRFunction birFunc = new BIRFunction(parentFunc.pos, funcName, funcName, 0, type,
+                DEFAULT_WORKER_NAME, 0, SymbolOrigin.VIRTUAL);
```

Comment:
Can't we use parentFunc.workerName here?

Reply:
Actually this worker name change had not effect because we don't have channels inside an array or map. But I changed this for accuracy. Since the new split function we are creating is a regular Ballerina function, I put the `DEFAULT_WORKER_NAME` for them as done in the others. Using `parentFunc.workerName` will be incorrect since the split function is not a worker and it will be wrong for splits inside workers.